pip/req/req_set.py

```diff
@@ -396,17 +397,26 @@ def _check_skip_installed(self, req_to_install, finder):
         # Check whether to upgrade/reinstall this req or not.
         req_to_install.check_if_exists()
         if req_to_install.satisfied_by:
-            skip_reason = 'satisfied (use --upgrade to upgrade)'
-            if self.upgrade:
+            upgrade_allowed = False
+
+            # Determine why upgrading this may not allowed.
+            if req_to_install.is_direct:
+                upgrade_allowed = True
+                skip_reason = 'already satisfied'
+            else:
+                skip_reason = 'skipped as not directly required'
```

Comment:
The logic for these messages is a bit confusing and general/vague comment: it's worth thinking about whether it could be refractored into one place instead of being spread out.

For this message in particular: I think this message is only shown for out-of-date dependencies that are none the less being skipped? If so it would be nice to make it something like "out pf date, but no upgrade requested". Or if it doesn't actually mean that, then "already satisfied, and no upgrade requested" would work better I think.